### PR TITLE
bugfix: fix duplicated resource id in PostgreSQL

### PR DESCRIPTION
**Pull request by l81893521**
### Ⅰ. Describe what this PR did
In this case
```
jdbc:postgresql://127.0.0.1:5432/seata?currentSchema=public
jdbc:postgresql://127.0.0.1:5432/seata?currentSchema=seata
```

it will cause the same resource id. and the error below occur.
1.Get file lock fail.
2.Cause the wrong table meta in pg sql.

### Ⅱ. Does this pull request fix one issue?
#2623

### CODE REVIEW

File: rm-datasource/src/main/java/io/seata/rm/datasource/DataSourceProxy.java [hunk 1]
```diff
@@ -141,7 +141,32 @@ public String getResourceGroupId() {
...
+             * 1.get file lock fail
+             * 2.error table meta cache
+             */
+            StringBuilder paramsBuilder = new StringBuilder();
+            String paramUrl = jdbcUrl.substring(jdbcUrl.indexOf('?') + 1, jdbcUrl.length());
+            String[] urlParams = paramUrl.split("&");
+            for (String urlParam : urlParams) {
+                if (urlParam.contains("currentSchema")) {
+                    paramsBuilder.append(urlParam);
+                }
```
Comment (zjinlei) [inline rm-datasource/src/main/java/io/seata/rm/datasource/DataSourceProxy.java:161]: 1.currentSchema is magic number
2.break?

Comment (l81893521) [reply rm-datasource/src/main/java/io/seata/rm/datasource/DataSourceProxy.java:162]: ok. Already add break

File: rm-datasource/src/main/java/io/seata/rm/datasource/DataSourceProxy.java [hunk 2]
```diff
@@ -141,7 +141,33 @@ public String getResourceGroupId() {
     @Override
     public String getResourceId() {
         if (jdbcUrl.contains("?")) {
-            return jdbcUrl.substring(0, jdbcUrl.indexOf('?'));
+            StringBuilder jdbcUrlBuilder = new StringBuilder();
```
Comment (zjinlei) [inline rm-datasource/src/main/java/io/seata/rm/datasource/DataSourceProxy.java:144]: According to dbType, non-pg remains the same

File: rm-datasource/src/main/java/io/seata/rm/datasource/DataSourceProxy.java [hunk 3]
```diff
@@ -141,7 +141,33 @@ public String getResourceGroupId() {
...
+                }
+            }
+
+            if (paramsBuilder.length() > 0) {
+                jdbcUrlBuilder.append("?");
+                jdbcUrlBuilder.append(paramsBuilder);
+            }
+
+            return jdbcUrlBuilder.toString();
         } else {
```
Comment (jsbxyyx) [inline rm-datasource/src/main/java/io/seata/rm/datasource/DataSourceProxy.java:171]: remove else, direct return?

File: rm-datasource/src/main/java/io/seata/rm/datasource/DataSourceProxy.java (hunk 2 above)
Comment (l81893521) [reply rm-datasource/src/main/java/io/seata/rm/datasource/DataSourceProxy.java:145]: fixed

File: rm-datasource/src/main/java/io/seata/rm/datasource/DataSourceProxy.java (hunk 3 above)
Comment (l81893521) [reply rm-datasource/src/main/java/io/seata/rm/datasource/DataSourceProxy.java:174]: I think
```
If() {

} else {

}
```
is more readable.

File: rm-datasource/src/main/java/io/seata/rm/datasource/DataSourceProxy.java [hunk 4]
```diff
@@ -141,7 +142,35 @@ public String getResourceGroupId() {
     @Override
     public String getResourceId() {
         if (jdbcUrl.contains("?")) {
-            return jdbcUrl.substring(0, jdbcUrl.indexOf('?'));
```
Comment (zjinlei) [inline rm-datasource/src/main/java/io/seata/rm/datasource/DataSourceProxy.java:144]: Keep the original clear, suggest:
return DBType.POSTGRESQL.name().equalsIgnoreCase(getDbType())? getPgResourceId() : jdbcUrl.substring(0,jdbcUrl.indexOf('?'));

Comment (l81893521) [reply rm-datasource/src/main/java/io/seata/rm/datasource/DataSourceProxy.java:144]: fixed.

Comment (zjinlei) [reply rm-datasource/src/main/java/io/seata/rm/datasource/DataSourceProxy.java:144]: Only separate the different parts.
if (!jdbcUrl.contains("?")) is the same

Comment (l81893521) [reply rm-datasource/src/main/java/io/seata/rm/datasource/DataSourceProxy.java:144]: as you say, the code would be like this 
```
if (jdbcUrl.contains("?")) {
    if (DBType.POSTGRESQL.name().equalsIgnoreCase(getDbType())) {
        getPGResourceId();
    } else {
      do something....
    }
} else {
    do something
}
```
the code is less, but the logic is more complicated.

**Pull request merged**